tests/tensor/test_subtensor.py

```diff
@@ -2713,41 +2713,43 @@ def test_static_shapes(x_shape, indices, expected):
     assert y.type.shape == expected
 
 
-def test_vectorize_subtensor_without_batch_indices():
-    signature = "(t1,t2,t3),()->(t1,t3)"
+@pytest.mark.parametrize(
+    "adv_idx, batch_idx",
+    [
+        (False, False),
+        (False, True),
+        (True, False),
+        # Advanced indexing with batched indexes fails when there are Slices / None
+        pytest.param(True, True, marks=pytest.mark.xfail(raises=NotImplementedError)),
+    ],
+)
+@config.change_flags(cxx="")  # C code not needed
+def test_vectorize_subtensor(adv_idx, batch_idx):
+    if adv_idx:
+        signature = "(t1,t2,t3),(idx)->(t1,tx,t3)"
+    else:
+        signature = "(t1,t2,t3),()->(t1,t3)"
 
-    def core_fn(x, start):
-        return x[:, start, :]
+    def core_fn(x, idx):
+        return x[:, idx, :]
```

Comment:
Do you want to leave room to add a test for the two weird advanced subindex conditions from numpy? Something like `x[idx1, :, idx2]` and `x[:, idx1, idx2, :]`?

The signature there would have to change to something like `"(t1, t2, t3), (t4), (t4)->(t2, t4)"` or `"(t1, t2, t3, t4), (t5), (t5)->(t2, t5, t4)"`.

Reply:
Added a bunch of tests. Thanks for the suggestion, it actually unconvered a bug with the non-contiguous advanced indexing